src/Emoji.ts

```diff
@@ -0,0 +1,24 @@
+// This function returns whether the first code point of the provided string is
+// within a range that contains emoji. Some of the codepoints in these ranges
+// are unassigned so this test is an approximation to the real thing.
+// Ranges taken from: https://stackoverflow.com/questions/30757193/
+// For a complete list of all assigned codepoints, go to:
+//   https://unicode.org/Public/emoji/11.0/emoji-data.txt
+export function isEmoji(str: string): boolean {
+  const codePoint = str.codePointAt(0);
+  if ((codePoint >= 0x1F600 && codePoint <= 0x1F64F) || // Emoticons
```

Comment:
You should short circuit this with something like:

```ts
if (codePoint < 8400) {
  return false;
}
```

Otherwise all the conditions run for every char